nbt/src/main/java/net/kyori/adventure/nbt/TagStringReader.java

```diff
@@ -24,6 +24,7 @@
 package net.kyori.adventure.nbt;
 
 import java.util.ArrayList;
+import java.util.HashMap;
```

Comment:
```suggestion

```
unused import (thanks checkstyle!)

apart from that, seems good. if you feel motivated, it'd be good to do similar tests for empty lists + arrays, but i don't mind doing those.

Reply:
Not sure how empty lists/arrays are handled by MC itself, but in adventure, `[]` becomes a list with one entry being an empty string (so `[""]`) and parsing `[L;]` fails - but I'd rather have you look at these :p

Reply:
ah -- i think both of those are incorrect behaviours :p